lib/radiator_web/controllers/api/authentication_controller.ex

```diff
@@ -0,0 +1,34 @@
+defmodule RadiatorWeb.Api.AuthenticationController do
+  use RadiatorWeb, :controller
+
+  alias Radiator.Auth
+  alias RadiatorWeb.Helpers.AuthHelpers
+
+  def create(conn, %{"name" => name, "password" => password}) do
+    Auth.Register.get_user_by_credentials(name, password)
+    |> case do
+      nil ->
+        send_resp(conn, 401, "Login failed")
+
+      user ->
+        token = Auth.Guardian.api_session_token(user)
+
+        conn
+        |> json(AuthHelpers.session_response(user, token))
+    end
+  end
+
+  def prolong(conn, _params) do
+    user = current_user(conn)
+    token = Auth.Guardian.api_session_token(user)
+
+    conn
+    |> json(AuthHelpers.session_response(user, token))
+  end
+
+  def logout(conn, _params) do
+    conn
+    |> Auth.Guardian.Plug.sign_out()
```

Comment:
sign out doesn't make much sense currently in the api, as the token just expires and we don't store any state. For the UI it is nice to remove it from the session. However, imho the rest api should not be stateful imho.